Objects/dictobject.c

```diff
@@ -2317,6 +2317,37 @@ PyDict_GetItemRef(PyObject *op, PyObject *key, PyObject **result)
     return _PyDict_GetItemRef_KnownHash(op, key, hash, result);
 }
 
+int
+_PyDict_GetItemRef_LockHeld(PyObject *op, PyObject *key, PyObject **result)
+{
+    ASSERT_DICT_LOCKED(op);
+    assert(PyUnicode_CheckExact(key));
```

Comment:
This seems a bit strange in that the unicode requirement is not indicated by the function name and not enforced by the similarly named `_PyDict_SetItem_LockHeld`.